### ISSUE TITLE
Plans to update to Spring Boot 3.0.0?

### ISSUE THREAD

**Issue by mhalbritter** [wavefrontHQ/wavefront-spring-boot#171]
Hi!

Are there plans to update the starter for Spring Boot 3.0? I just tried it against Spring Boot 3.0.0 SNAPSHOT and it fails with:

```
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'wavefrontMeterRegistry' defined in class path resource [org/springframework/boot/actuate/autoconfigure/metrics/export/wavefront/WavefrontMetricsExportAutoConfiguration.class]: Unsatisfied dependency expressed through method 'wavefrontMeterRegistry' parameter 2: Error creating bean with name 'wavefrontSender' defined in class path resource [org/springframework/boot/actuate/autoconfigure/wavefront/WavefrontSenderConfiguration.class]: Failed to instantiate [com.wavefront.sdk.common.WavefrontSender]: Property management.wavefront.api-token with value 'null' is invalid: This property is mandatory whenever publishing directly to the Wavefront API
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:774) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:524) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1323) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1160) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:566) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:526) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:326) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:324) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:254) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1374) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1294) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:861) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:765) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	... 53 common frames omitted
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'wavefrontSender' defined in class path resource [org/springframework/boot/actuate/autoconfigure/wavefront/WavefrontSenderConfiguration.class]: Failed to instantiate [com.wavefront.sdk.common.WavefrontSender]: Property management.wavefront.api-token with value 'null' is invalid: This property is mandatory whenever publishing directly to the Wavefront API
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:633) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:621) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1323) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1160) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:566) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:526) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:326) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:324) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:254) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1374) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1294) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:861) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:765) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	... 67 common frames omitted
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.wavefront.sdk.common.WavefrontSender]: Property management.wavefront.api-token with value 'null' is invalid: This property is mandatory whenever publishing directly to the Wavefront API
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:161) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:629) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	... 81 common frames omitted
Caused by: org.springframework.boot.context.properties.source.InvalidConfigurationPropertyValueException: Property management.wavefront.api-token with value 'null' is invalid: This property is mandatory whenever publishing directly to the Wavefront API
	at org.springframework.boot.actuate.autoconfigure.wavefront.WavefrontProperties.getApiTokenOrThrow(WavefrontProperties.java:126) ~[spring-boot-actuator-autoconfigure-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.springframework.boot.actuate.autoconfigure.wavefront.WavefrontSenderConfiguration.wavefrontSender(WavefrontSenderConfiguration.java:50) ~[spring-boot-actuator-autoconfigure-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:568) ~[na:na]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:130) ~[spring-beans-6.0.0-20220801.162650-670.jar:6.0.0-SNAPSHOT]
	... 82 common frames omitted
```

In `AccountManagementEnvironmentPostProcessor` there are some property names which have to be updated, as Spring Boot moved them.

**Comment by snicoll** [comment]
WIP in this PR https://github.com/wavefrontHQ/wavefront-spring-boot/pull/167

**Comment by oppegard** [comment]
`wavefront-spring-boot` 3.0.x has been released with Boot 3.0.0 compatibility: https://github.com/wavefrontHQ/wavefront-spring-boot/releases/tag/v3.0.1. This is on the `springboot3` branch, while the Boot 2.x work remains on `master`. The `springboot3` branch has been made the default branch in the Github repo configuration as well.
